lib/http/client.cpp

```diff
@@ -962,6 +962,25 @@ Client::send_to_device(const std::string &event_type,
         put<nlohmann::json>(api_path, body, callback);
 }
 
+void
+Client::post_public_rooms(const std::string &server, const nhlomann::json &j, 
+                                Callback<mtx::responses::PublicRooms> cb)
+{
+        const auto api_path = "/client/r0/publicRooms" + " " + server;
```

Comment:
This is wrong, you need to urlencode parameters, spaces are invalid in api paths, etc. Check the other functions using `mtx::client::utils::url_encode` and `mtx::client::utils::query_params`.